api/internal/loader/fileloader.go

```diff
@@ -318,11 +321,19 @@ func (fl *FileLoader) httpClientGetContent(path string) ([]byte, error) {
 	defer resp.Body.Close()
 	// response unsuccessful
 	if resp.StatusCode < 200 || resp.StatusCode > 299 {
-		_, err = git.NewRepoSpecFromURL(path)
-		if err == nil {
-			return nil, errors.Errorf("URL is a git repository")
+		if resp.StatusCode == MultipleChoicesRedirectCode {
+			var newPath string = resp.Header.Get("Location")
+			return nil, &RedirectionError{
+				Msg:     "Response is redirect",
+				NewPath: newPath,
+			}
```

Comment:
Maybe That is better to solve redirects here than return `RedirectionError` and a new path.

Reply:
Hi! It does not work in all cases. For example if I return redirect to git. In that case httpClient has noe credentials for download content

Reply:
> For example if I return redirect to git. In that case httpClient has no credentials for download content

Sorry, I couldn't understand about this sentence.
In my understanding, that repo uses the `HTTPS` protocol to download when HTTP redirects to git.

Reply:
My explanation was not clear. Sorry for that. I'll try again.
Example
We get next url in resource section `https://example.com/apps/app1?version=1.1.0`
When kustomize (in build  proccess) go to that url, it get `Redirect 300` with `newPath` `https://customgit.org/somegrop/project-app1.git?refs=1.1.0`. The repository is PRIVATE.
In this case if we go to that git repo using `httpClient` we get page with ask for credential (((
But if we create a new loader kustomize will use console `git` **which already have credentials and can clone repository**